checks.d/mysql.py

```diff
@@ -803,21 +812,54 @@ def _is_innodb_engine_enabled(self, db):
             return return_val
 
     def _get_replica_stats(self, db):
-        with closing(db.cursor()) as cursor:
-            cursor.execute("SHOW SLAVE STATUS;")
-            replica_results = dict(cursor.fetchall())
+        try:
+            with closing(db.cursor()) as cursor:
+                cursor.execute("SHOW SLAVE STATUS;")
+                replica_results = dict(cursor.fetchall())
+                cursor.execute("SHOW MASTER STATUS;")
+                binlog_results = dict(cursor.fetchall())
+                if binlog_results:
+                    replica_results.update({'Binlog_enabled': True})
+
+                return replica_results
+        except (pymysql.err.InternalError, pymysql.err.OperationalError) as e:
+            self.log.warning("Privileges error getting replication status (must grant REPLICATION CLIENT): %s", e)
```

Comment:
Could you use `self.warning` instead? On top of logging at the `WARNING` level it will add the message to the `info` page and also show up as an "integration error" on the web UI.